src/Http/Http.Features/src/IHeaderDictionary.Keyed.cs

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Http
+{
+    public partial interface IHeaderDictionary
```

Comment:
Yes; I had a source generator for the IHeaderDictionary (how I originally generated them); but the CI rejected it as a used project that wasn't in the shipped API, which seemed to be going wrong since it was just internal and not to be shipped.

Could automate it as part of the Kestrel code generator (so only need to add to HeaderNames); however that would need an attribute on some of the headers (e.g. `DNT`) to say don't add them (Obselete may work or may be too strong; since it also has `Path`,`Scheme`, `Method` etc.)